drivers/npmx/npmx_shell.c

```diff
@@ -1792,6 +1799,165 @@ static int cmd_adc_meas_take_vbat(const struct shell *shell, size_t argc, char *
 	return 0;
 }
 
+npmx_timer_mode_t timer_mode_int_convert_to_enum(uint32_t timer_mode)
+{
+	switch (timer_mode) {
+	case 0:
+		return NPMX_TIMER_MODE_BOOT_MONITOR;
+	case 1:
+		return NPMX_TIMER_MODE_WATCHDOG_WARNING;
+	case 2:
+		return NPMX_TIMER_MODE_WATCHDOG_RESET;
+	case 3:
+		return NPMX_TIMER_MODE_GENERAL_PURPOSE;
+	case 4:
+		return NPMX_TIMER_MODE_WAKEUP;
+	default:
+		return NPMX_TIMER_MODE_INVALID;
+	}
+}
+
+static int timer_config_get(const struct shell *shell, npmx_timer_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	npmx_timer_t *timer_instance = npmx_timer_get(npmx_instance, 0);
+
+	npmx_timer_config_t timer_config;
+	npmx_error_t err_code = npmx_timer_config_get(timer_instance, &timer_config);
+
+	if (check_error_code(shell, err_code)) {
+		if (config_type == NPMX_TIMER_CONFIG_TYPE_MODE) {
+			shell_print(shell, "Value: %d.", timer_config.mode);
+		} else if (config_type == NPMX_TIMER_CONFIG_TYPE_PRESCALER) {
+			shell_print(shell, "Value: %d.", timer_config.prescaler);
+		} else if (config_type == NPMX_TIMER_CONFIG_TYPE_COMPARE) {
+			shell_print(shell, "Value: %d.", timer_config.compare_value);
+		}
+	} else {
+		shell_error(shell, "Error: unable to read timer configuration.");
+	}
+
+	return 0;
+}
+
+static int timer_config_set(const struct shell *shell, size_t argc, char **argv,
+			    npmx_timer_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	char config_name[10];
+	if (config_type == NPMX_TIMER_CONFIG_TYPE_MODE) {
+		strcpy(config_name, "mode");
+	} else if (config_type == NPMX_TIMER_CONFIG_TYPE_PRESCALER) {
+		strcpy(config_name, "prescaler");
+	} else if (config_type == NPMX_TIMER_CONFIG_TYPE_COMPARE) {
+		strcpy(config_name, "period");
+	} else {
+		shell_error(shell, "Error: invalid config type value.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing timer %s value.", config_name);
+		return 0;
+	}
+
+	int err = 0;
+	uint32_t input_value = CLAMP(shell_strtoul(argv[1], 0, &err), 0, 0xFFFFFFUL);
```

Comment:
I am not sure about this. We already had a little discussion [here](https://github.com/NordicSemiconductor/npmx-zephyr/pull/24#discussion_r1244009996) about clamping to max value instead of `UINT8_MAX` or something like that, and it was decided we should not do that, because for input greater than `0xFFFFFFUL` it will always return `0xFFFFFFUL`. It should rather be checked before assigning to `compare_value `, if it is not too big.